### PR TITLE
cmake: 3.24.3 -> 3.25.1

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Have I said how I hate such errors on Darwin?

**Review by winterqt — CHANGES_REQUESTED** [review]
Instead of commenting out the lines, can you remove them like the previous patch does ([ref](https://github.com/NixOS/nixpkgs/blob/997d9f10ad1e4ed8f2836ad3eeb0bbfa3b43c2e4/pkgs/development/tools/build-managers/cmake/002-application-services.diff))?

File: pkgs/development/tools/build-managers/cmake/002-application-services.diff [hunk 1]
```diff
@@ -1,20 +1,18 @@
-diff --git a/Source/CMakeLists.txt b/Source/CMakeLists.txt
```
Comment (winterqt) [inline pkgs/development/tools/build-managers/cmake/002-application-services.diff:1]: To reduce the diff, can you clone the CMake repo, checkout the 3.25.1 tag, remove the lines, and then `git diff`? (I'm sorry for being so pedantic here, please yell at me if you think I should just stop.)

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/build-managers/cmake/002-application-services.diff (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/development/tools/build-managers/cmake/002-application-services.diff:1]: No problem. I am usually way more pedantic.

However, I do not generate patches by using the git repo, but the tarball provided by them. I will take a bit more time.

**Comment by AndersonTorres** [comment]
If this is working on Darwin ARM, can I be confident this is working at all?

File: pkgs/development/tools/build-managers/cmake/002-application-services.diff (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/development/tools/build-managers/cmake/002-application-services.diff:1]: Seriously, I will refrain from this. The patch is working as expected, and this is too much work for a "beautiful" patch.

Comment (winterqt) [reply pkgs/development/tools/build-managers/cmake/002-application-services.diff:1]: I'll do it then, if you give me a (verbal) OK to force push.

**Comment by winterqt** [comment]
> If this is working on Darwin ARM, can I be confident this is working at all?

What do you mean? Like, have you only tested it on Darwin, and you're making sure if it's worth testing anywhere else if it works there? A bit confused as to what you're asking :)

**Comment by AndersonTorres** [comment]
Specifically, the Darwin X86 is failing because of some timeout building LLVM; so, this is not directly related to Cmake.

The ofborg builds on the other three platforms - two Linuces and the Darwin ARM - worked well.

**Comment by Mindavi** [comment]
Darwin x86 and arm64 use different toolchains, so they may differ. But in general I'd say that 3 working platforms gives good confidence. Some reverse deps should be built before merging to be sure though.

File: pkgs/development/tools/build-managers/cmake/002-application-services.diff (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/development/tools/build-managers/cmake/002-application-services.diff:1]: Many thanks, @winterqt! I have solved it.

And merry nixmas!

**Comment by AndersonTorres** [comment]
@ofborg build cmakeMinimal

**Comment by AndersonTorres** [comment]
@ofborg build cmakeMinimal pkgsCross.x86_64-netbsd.cmakeMinimal

**Comment by AndersonTorres** [comment]
@ofborg build cmakeMinimal pkgsCross.x86_64-netbsd.cmakeMinimal pkgsCross.x86_64-darwin.cmakeMinimal

**Comment by AndersonTorres** [comment]
@ofborg build cmakeMinimal  pkgsCross.x86_64-darwin.cmakeMinimal

**Comment by winterqt** [comment]
I don't think aarch64 -> x86_64 Darwin cross works at all, FWIW.

**Comment by AndersonTorres** [comment]
> I don't think aarch64 -> x86_64 Darwin cross works at all, FWIW.

Damn! It would be a cool test suite!

**Comment by AndersonTorres** [comment]
The same error repeats:

> error: building of '/nix/store/qphaxwdxa26y9dq05306x7faxys8fbqb-llvm-11.1.0.drv!lib,out' from .drv file timed out after 3600 seconds

**Comment by AndersonTorres** [comment]
@ofborg build cmakeMinimal

**Comment by AndersonTorres** [comment]
https://github.com/ofborg build cmakeMinimal rapidjson

**Comment by AndersonTorres** [comment]
@ofborg build cmakeMinimal

**Comment by AndersonTorres** [comment]
@ofborg build cmakeMinimal

**Comment by AndersonTorres** [comment]
I am cogitating to split this into old and new releases.

**Comment by AndersonTorres** [comment]
@ofborg build rapidjson cmakeMinimal starship

**Comment by AndersonTorres** [comment]
@ofborg build cmakeMinimal starship

**Pull request merged**